TST: fitsio cannot build, blocking the parallel job

CI started failing this morning from some cfitsio build error. 

Example log: https://github.com/astropy/astropy/actions/runs/11293862417/job/31413011223

https://github.com/astropy/astropy/blob/21181db54df97f7b7edb7b2142d929e33d5a455b/.github/workflows/ci_workflows.yml#L69-L77

https://github.com/astropy/astropy/blob/21181db54df97f7b7edb7b2142d929e33d5a455b/tox.ini#L29

https://github.com/astropy/astropy/blob/21181db54df97f7b7edb7b2142d929e33d5a455b/tox.ini#L74

```
Get:1 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 language-pack-fr-base all 1:24.04+20240817 [3157 kB]
Get:2 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 language-pack-fr all 1:24.04+20240817 [1900 B]
Get:3 http://azure.archive.ubuntu.com/ubuntu noble/universe amd64 libcfitsio10t64 amd64 4.3.1-1.1build2 [528 kB]
Get:4 http://azure.archive.ubuntu.com/ubuntu noble/universe amd64 libcfitsio-dev amd64 4.3.1-1.1build2 [598 kB]
Get:5 http://azure.archive.ubuntu.com/ubuntu noble/universe amd64 libcfitsio-doc all 4.3.1-1.1build2 [2076 kB]
...
Collecting numpy (from fitsio)
Using cached numpy-2.1.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
Using cached numpy-2.1.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.0 MB)
Building wheels for collected packages: fitsio
  Building wheel for fitsio (pyproject.toml): started
  Building wheel for fitsio (pyproject.toml): finished with status 'error'
Failed to build fitsio
...
configure: error: Unable to locate bz2 library needed when enabling bzip2 support; try specifying the path
```

p.s. @olebole is this something you are aware of?

--- a/.github/workflows/ci_workflows.yml
+++ b/.github/workflows/ci_workflows.yml
@@ -72,6 +72,7 @@ jobs:
             apt:
               - language-pack-fr
               - tzdata
+              - libbz2-dev
               - libcfitsio-dev
           toxargs: -v --develop
           posargs: -n=4 --run-slow

